localstack/services/awslambda/invocation/runtime_executor.py

```diff
@@ -0,0 +1,233 @@
+import logging
+import re
+import shutil
+import time
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import Dict, Literal, Optional, Tuple
+
+from localstack import config
+from localstack.services.awslambda.invocation.executor_endpoint import (
+    ExecutorEndpoint,
+    ServiceEndpoint,
+)
+from localstack.services.awslambda.invocation.lambda_models import FunctionVersion
+from localstack.services.awslambda.lambda_utils import (
+    get_container_network_for_lambda,
+    get_main_endpoint_from_container,
+)
+from localstack.services.install import LAMBDA_RUNTIME_INIT_PATH
+from localstack.utils.archives import unzip
+from localstack.utils.container_utils.container_client import ContainerConfiguration
+from localstack.utils.docker_utils import DOCKER_CLIENT as CONTAINER_CLIENT
+from localstack.utils.net import get_free_tcp_port
+
+LOG = logging.getLogger(__name__)
+
+RUNTIME_REGEX = r"(?P<runtime>[a-z]+)(?P<version>\d+(\.\d+)?(\.al2)?)(?:.*)"
+
+# IMAGE_PREFIX = "gallery.ecr.aws/lambda/"
+IMAGE_PREFIX = "amazon/aws-lambda-"
+
+RAPID_ENTRYPOINT = "/var/rapid/init"
+
+InitializationType = Literal["on-demand", "provisioned-concurrency"]
+
+LAMBDA_DOCKERFILE = """FROM {base_img}
+COPY aws-lambda-rie {rapid_entrypoint}
+COPY code/ /var/task
+"""
+
+
+# TODO provided runtimes
+# TODO a tad hacky, might cause problems in the future.. just use mapping?
+def get_runtime_split(runtime: str) -> Tuple[str, str]:
+    match = re.match(RUNTIME_REGEX, runtime)
+    if match:
+        runtime, version = match.group("runtime"), match.group("version")
+        # sad exception for .net
+        if runtime == "dotnetcore":
+            runtime = "dotnet"
+            version = f"core{version}"
+        return runtime, version
+    raise Exception("Cannot process runtime '%s'" % runtime)
```

Comment:
nit: raise `ValueError` instead and maybe `Unknown runtime '...` as message instead?